Lib/graphlib.py

```diff
@@ -248,3 +248,78 @@ def static_order(self):
             self.done(*node_group)
 
     __class_getitem__ = classmethod(GenericAlias)
+
+
+def reverse(graph):
+    """Reverse the direction of the edges in a directed graph.
+
+    Given a mapping from nodes to collections of their dependencies,
+    construct a dict mapping each node to the set of nodes that depend on it.
+
+    Nodes that have no dependents appear as keys in the result, with an empty
+    set as value.
+
+    For example:
+
+        >>> reverse({"a": ["b", "c"], "d": []})
+        {'b': {'a'}, 'c': {'a'}, 'a': set(), 'd': set()}
+
+    """
+    result = {}
+    for node, deps in graph.items():
+        for dep in deps:
+            if dep not in result:
+                result[dep] = {node}
+            else:
+                result[dep].add(node)
```

Comment:
is `setdefault` worth considering here?

```suggestion
            result.setdefault(dep, set()).add(node)
```

Reply:
No, because it allocates a `set` and discards it if the key is present. In performance sensitive code you should only use `.setdefault()` with constant values, or use `collections.defaultdict`.